src/pykeen/quadruples/quadruples_factory.py

```diff
@@ -0,0 +1,1246 @@
+    tail_filter = tail_column < 0
+    timestamp_filter = timestamp_column < 0
+    num_no_head = head_filter.sum()
+    num_no_relation = relation_filter.sum()
+    num_no_tail = tail_filter.sum()
+    num_no_timestamp = timestamp_filter.sum()
+
+    if (num_no_head > 0) or (num_no_relation > 0) or (num_no_tail > 0) or (num_no_timestamp > 0):
+        logger.warning(
+            f"You're trying to map quadruples with {num_no_head + num_no_tail} entities, {num_no_relation} relations "
+            f"and {num_no_timestamp} timestamps "
+            f"that are not in the training set. These quadruples will be excluded from the mapping.",
+        )
+        non_mappable_quadruples = head_filter | relation_filter | tail_filter | timestamp_filter
+        head_column = head_column[~non_mappable_quadruples, None]
+        relation_column = relation_column[~non_mappable_quadruples, None]
+        tail_column = tail_column[~non_mappable_quadruples, None]
+        timestamp_column = timestamp_column[~non_mappable_quadruples, None]
+        logger.warning(
+            f"In total {non_mappable_quadruples.sum():.0f} from {quadruples.shape[0]:.0f} quadruples were filtered out",
+        )
+    quadruples_of_ids = np.concatenate(
+        [head_column, relation_column, tail_column, timestamp_column], axis=1
+    )
+    quadruples_of_ids = np.array(quadruples_of_ids, dtype=np.int64)
+    unique_mapped_quadruples = np.unique(ar=quadruples_of_ids, axis=0)
+    return torch.tensor(unique_mapped_quadruples, dtype=torch.long)
+
+
+class CoreQuadruplesFactory(KGInfo):
+    """Create instances from ID-based quadruples."""
+
+    def __init__(
+        self,
+        mapped_quadruples: Union[MappedQuadruples, np.ndarray],
+        num_entities: int,
+        num_relations: int,
+        num_timestamps: int,
+        create_inverse_quadruples: bool = False,
+        metadata: Optional[Mapping[str, Any]] = None,
+    ):
+        """
+        Create the quadruples factory.
+
+        :param mapped_quadruples: shape: (n, 4)
+            A four-column matrix where each row are the head identifier, relation identifier, tail identifier, then timestamp.
+        :param num_entities:
+            The number of entities.
+        :param num_relations:
+            The number of relations.
+        :param num_timestamps:
+            The number of timestamps.
+        :param create_inverse_quadruples:
+            Whether to create inverse quadruples.
+        :param metadata:
+            Arbitrary metadata to go with the graph.
+
+        :raises TypeError:
+            if the mapped_quadruples are of non-integer dtype
+        :raises ValueError:
+            if the mapped_quadruples are of invalid shape
+        """
+        super().__init__(
+            num_entities=num_entities,
+            num_relations=num_relations,
+            create_inverse_triples=create_inverse_quadruples,
+        )
+        self.create_inverse_quadruples = create_inverse_quadruples
```

Comment:
Why do you define a new constant when it is exactly the same as `self.create_inverse_triples`?

Reply:
Yes it's unnecessary duplication. I removed the redundant parameter in the new commit